packages/svelte/src/internal/client/reactivity/batch.js

```diff
@@ -444,6 +418,51 @@ export class Batch {
 	static enqueue(task) {
 		queue_micro_task(task);
 	}
+
+	/**
+	 * @param {Batch} current_batch
+	 */
+	static apply(current_batch) {
+		if (!async_mode_flag || batches.size === 1) {
+			return noop;
+		}
+
+		// if there are multiple batches, we are 'time travelling' —
+		// we need to undo the changes belonging to any batch
+		// other than the current one
+
+		/** @type {Map<Source, { v: unknown, wv: number }>} */
+		var current_values = new Map();
+		batch_deriveds = new Map();
+
+		for (const [source, current] of current_batch.current) {
+			current_values.set(source, { v: source.v, wv: source.wv });
+			source.v = current;
+		}
+
+		for (const batch of batches) {
+			if (batch === current_batch) continue;
+
+			for (const [source, previous] of batch.#previous) {
+				if (!current_values.has(source)) {
+					current_values.set(source, { v: source.v, wv: source.wv });
+					source.v = previous;
+				}
+			}
+		}
+
+		return () => {
+			for (const [source, { v, wv }] of current_values) {
+				// reset the source to the current value (unless
+				// it got a newer value as a result of effects running)
+				if (source.wv <= wv) {
```

Comment:
it's gonna be funny the day we somehow overflow this

Reply:
if you updated a source once a millisecond you would overflow in...

```js
const ONE_YEAR = 365 * 24 * 60 * 60 * 1000;
Number.MAX_SAFE_INTEGER / ONE_YEAR;
```

...285,616 years